### PR TITLE
refactor!: avoid unnecessary copying in normalization

### PR DESCRIPTION
**Pull request by tobymao**

### CODE REVIEW

File: sqlglot/schema.py [hunk 1]
```diff
@@ -451,16 +447,17 @@ def normalize_name(
...
-) -> str:
+) -> exp.Identifier:
     if isinstance(identifier, str):
         identifier = exp.parse_identifier(identifier, dialect=dialect)
 
-    if not normalize:
-        return identifier.name
-
     # This can be useful for normalize_identifier
     identifier.meta["is_table"] = is_table
```
Comment (georgesittas) [inline sqlglot/schema.py:455]: I think this needs to be moved after the following branch?

Comment (tobymao) [reply sqlglot/schema.py:455]: why?

Comment (georgesittas) [reply sqlglot/schema.py:455]: Because IIRC it's only needed for normalizations, right? If we won't normalize then why should we mutate here?

**Review by georgesittas — APPROVED** [review]
Seems good, I understand the "unnecessary" copying refers to the previous `_normalize_table` implementation, right? So now we only copy if we need to normalize?

**Comment by tobymao** [comment]
right, also converting identifiers to string and reparsing them

**Pull request merged**